storage/src/test/java/org/apache/kafka/storage/internals/checkpoint/PartitionMetadataFileTest.java

```diff
@@ -0,0 +1,98 @@
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.errors.InconsistentTopicIdException;
+
+import org.apache.kafka.storage.internals.log.LogDirFailureChannel;
+import org.apache.kafka.test.TestUtils;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PartitionMetadataFileTest  {
+    private final File dir = TestUtils.tempDirectory();
+
+    @Test
+    public void testSetRecordWithDifferentTopicId() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        Uuid differentTopicId = Uuid.randomUuid();
+        assertThrows(InconsistentTopicIdException.class, () -> partitionMetadataFile.record(differentTopicId));
+    }
+
+    @Test
+    public void testSetRecordWithSameTopicId() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+    }
+
+    @Test
+    public void testMaybeFlushWithTopicIdPresent() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        assertDoesNotThrow(partitionMetadataFile::maybeFlush);
+
+        assertDoesNotThrow(() -> {
+            List<String> lines = Files.readAllLines(file.toPath());
+            assertEquals(2, lines.size());
+            assertEquals("version: 0", lines.get(0));
+            assertEquals("topic_id: " + topicId, lines.get(1));
+        });
+    }
+
+    @Test
+    public void testMaybeFlushWithNoTopicIdPresent() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+
+        assertDoesNotThrow(partitionMetadataFile::maybeFlush);
+        assertEquals(0, file.length());
+    }
+
+    @Test
+    public void testRead() {
+        File file = PartitionMetadataFile.newFile(dir);
+        LogDirFailureChannel channel = Mockito.mock(LogDirFailureChannel.class);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, channel);
+
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
```

Comment:
we don't need to use `assertDoesNotThrow` since it is fine to add any exception to method signature.

Reply:
Oh, in here the reason I used `assertDoesNotThrow` is because we expect the first time to `record` the `topicId` shouldn't generate any exception, since there's no `dirtyTopicIdOpt`

Do you think that make sense? Or that would be redundant?

Reply:
The only exception would be generated is `InconsistentTopicIdException` in this case I think

Reply:
>I used assertDoesNotThrow is because we expect the first time to record the topicId shouldn't generate any exception, since there's no dirtyTopicIdOpt

I agree to that. My point was `assertDoesNotThrow` converts the checked exception to unchecked exception, and it is useful when we run assert in the lambda. However, in those test case we can add the exception to method signature, and just let it fail. 

In short, `assertDoesNotThrow` in those test cases is redundant wrapper.

Reply:
I see, thank you very much for the detailed explanation! I've revised it as suggested.